library/core/src/hash/sip.rs

```diff
@@ -138,7 +138,7 @@ unsafe fn u8to64_le(buf: &[u8], start: usize, len: usize) -> u64 {
         out |= (unsafe { *buf.get_unchecked(start + i) } as u64) << (i * 8);
         i += 1;
     }
-    debug_assert_eq!(i, len);
+    debug_assert!(i == len);
```

Comment:
We should probably revert back for the error message when #103639 has landed. So please add a fixme.

Reply:
Should it be FIXME(fee1-dead) or FIXME(debug_assert_eq)?

Reply:
`fee1-dead` is fine.